splunk-quarkus/src/main/java/com/redhat/console/integrations/servicenow/ServiceNowIntegration.java

```diff
@@ -81,13 +82,17 @@ private void configureHandler() throws Exception {
                 .process(new BasicAuthenticationProcessor("rh_insights_integration"))
 
                 .setHeader(Exchange.HTTP_URI, header("targetUrl"))
-                .to(https("dynamic")
-                        .httpMethod("POST")
-                        .headerFilterStrategy(new ServiceNowHttpHeaderStrategy())
-                        .advanced()
-                        .httpClientConfigurer(getClientConfigurer()))
+                .to(seda("push"));
+    }
 
-                .to(direct("success"));
+    private void configurePush() throws Exception {
+        from(seda("push").concurrentConsumers(10))
```

Comment:
Indeed there is a limit which will raise an exception when reached. This would be an unhandled exception, which should fail the Exchange and stop Kafka offset commits. In that case, I expect the Kafka to resend messages, however some can be lost. We have other option to "wait when full", in that case we can end up in the same situation as we did before when the Kafka poll would not be done on time, leading to Kafka consumer being kicked out. Do you have a suggestion how to improve it even further?

Reply:
It may be possible to prevent the kick out by changing the Kafka config. We've done that in Notifications because we were sometimes kicked out from the consumer group.

| Config key | Default value | Notifications-engine value |
| - | - | - |
| [max.poll.interval.ms](https://docs.confluent.io/platform/current/installation/configuration/consumer-configs.html#max-poll-interval-ms)  | 300000 | 1200000 |
| [max.poll.records](https://docs.confluent.io/platform/current/installation/configuration/consumer-configs.html#max-poll-records) | 500 | 100 |

In Notifications, this is done with [SmallRye Reactive Messaging config keys](https://github.com/RedHatInsights/notifications-backend/blob/41ac032e469c30ab62cb311c7e47f950d4d3a054/.rhcicd/clowdapp-engine.yaml#L93-L96). Camel may have something similar.

Reply:
Yep, see #117